src/cred.rs

```diff
@@ -481,6 +481,12 @@ impl CredentialHelper {
     }
 }
 
+fn is_absolute_path(path: &str) -> bool {
+    path.starts_with('/')
+        || path.starts_with('\\')
+        || cfg!(windows) && path.chars().nth(1).is_some_and(|x| x == ':')
```

Comment:
Just a minor note, this is slightly different from https://github.com/git-for-windows/git/blob/cca1f38702730b35f52c29efd62864b85e85ddcc/compat/win32/path-utils.c#L9-L31, but probably not enough to matter here.

Reply:
Yes, `git2-rs` uses the `str` type while the git C uses a `char*` and if I understand the git C code correctly, it just tries to handle the case where the drive letter is a UTF8 character. I'd expect the `str::nth` function to take care of that already. However, there might indeed be some behaviour differences in corner cases at I'm not aware of.